### PR TITLE
DAOS-16500 build: Add support for Leap 15.6

### PR DESCRIPTION
**Pull request by phender**

### CODE REVIEW

File: vars/getChrootName.groovy [hunk 1]
```diff
@@ -14,7 +14,7 @@ String call(String distro) {
 
     return[ 'centos7':    'centos+epel-7-x86_64',
             'el8':        'rocky+epel-8-x86_64',
-            'leap15':     'opensuse-leap-15.3-x86_64',
+            'leap15':     'opensuse-leap-15.6-x86_64',
```
Comment (brianjmurrell) [inline vars/getChrootName.groovy:17]: Per our conversation, I think you are going to have to hold this back to 15.5.  See also https://github.com/rpm-software-management/mock/issues/1516.

```suggestion
            'leap15':     'opensuse-leap-15.5-x86_64',
```

File: vars/packageBuildingPipelineDAOSTest.groovy [hunk 2]
```diff
@@ -551,26 +551,26 @@ void call(Map pipeline_args) {
...
-                                additionalBuildArgs dockerBuildArgs() + '--build-arg FVERSION=37'
+                                additionalBuildArgs dockerBuildArgs() + '--build-arg FVERSION=41'
                             }
                         }
                         steps {
                             sh label: 'Build package',
                                script: '''rm -rf artifacts/leap15/
                                           mkdir -p artifacts/leap15/
-                                          make CHROOT_NAME="opensuse-leap-15.5-x86_64" ''' +
+                                          make CHROOT_NAME="opensuse-leap-15.6-x86_64" ''' +
```
Comment (brianjmurrell) [inline vars/packageBuildingPipelineDAOSTest.groovy:561]:
```suggestion
                                          make CHROOT_NAME="opensuse-leap-15.5-x86_64" ''' +
```

File: vars/packageBuildingPipelineDAOSTest.groovy [hunk 3]
```diff
@@ -551,26 +551,26 @@ void call(Map pipeline_args) {
...
                                               'DISTRO_VERSION=' + parseStageInfo()['distro_version'] + ' ' +
                                        pipeline_args.get('make args', '') + ' chrootbuild ' +
                                        pipeline_args.get('add_make_targets', '')
                         }
                         post {
                             success {
-                                rpmlintMockResults('opensuse-leap-15.5-x86_64',
+                                rpmlintMockResults('opensuse-leap-15.6-x86_64',
```
Comment (brianjmurrell) [inline vars/packageBuildingPipelineDAOSTest.groovy:568]:
```suggestion
                                rpmlintMockResults('opensuse-leap-15.5-x86_64',
```

File: vars/packageBuildingPipelineDAOSTest.groovy [hunk 4]
```diff
@@ -551,26 +551,26 @@ void call(Map pipeline_args) {
...
                                                    pipeline_args.get('rpmlint_rpms_allow_errors', false),
                                                    pipeline_args.get('rpmlint_rpms_skip', false),
                                                    pipeline_args.get('make args', ''))
                                 sh label: 'Collect artifacts',
-                                   script: '''(cd /var/lib/mock/opensuse-leap-15.5-x86_64/result/ &&
+                                   script: '''(cd /var/lib/mock/opensuse-leap-15.6-x86_64/result/ &&
```
Comment (brianjmurrell) [inline vars/packageBuildingPipelineDAOSTest.groovy:573]:
```suggestion
                                   script: '''(cd /var/lib/mock/opensuse-leap-15.5-x86_64/result/ &&
```

File: vars/packageBuildingPipelineDAOSTest.groovy [hunk 5]
```diff
@@ -587,7 +587,7 @@ void call(Map pipeline_args) {
                             }
                             unsuccessful {
                                 sh label: 'Build Log',
-                                   script: '''mockroot=/var/lib/mock/opensuse-leap-15.5-x86_64
+                                   script: '''mockroot=/var/lib/mock/opensuse-leap-15.6-x86_64
```
Comment (brianjmurrell) [inline vars/packageBuildingPipelineDAOSTest.groovy:590]:
```suggestion
                                   script: '''mockroot=/var/lib/mock/opensuse-leap-15.5-x86_64
```

File: vars/packageBuildingPipelineDAOSTest.groovy [hunk 6]
```diff
@@ -597,7 +597,7 @@ void call(Map pipeline_args) {
                             }
                             always {
                                 sh label: 'Collect config.log(s)',
-                                   script: '(if cd /var/lib/mock/opensuse-leap-15.5-x86_64/root/builddir/build/' +
+                                   script: '(if cd /var/lib/mock/opensuse-leap-15.6-x86_64/root/builddir/build/' +
```
Comment (brianjmurrell) [inline vars/packageBuildingPipelineDAOSTest.groovy:600]:
```suggestion
                                   script: '(if cd /var/lib/mock/opensuse-leap-15.5-x86_64/root/builddir/build/' +
```

File: vars/packageBuildingPipelineDAOSTest.groovy [hunk 7]
```diff
@@ -614,7 +614,7 @@ void call(Map pipeline_args) {
                                 archiveArtifacts artifacts: 'artifacts/leap15/**'
                             }
                         }
-                    } //stage('Build RPM on Leap 15')
+                    } //stage('Build RPM on Leap 15.6')
```
Comment (brianjmurrell) [inline vars/packageBuildingPipelineDAOSTest.groovy:617]:
```suggestion
                    } //stage('Build RPM on Leap 15.5')
```

File: vars/packageBuildingPipelineDAOSTest.groovy [hunk 8]
```diff
@@ -538,7 +538,7 @@ void call(Map pipeline_args) {
                             }
                         }
                     } //stage('Build RPM on EL 9')
-                    stage('Build RPM on Leap 15.5') {
+                    stage('Build RPM on Leap 15.6') {
```
Comment (brianjmurrell) [inline vars/packageBuildingPipelineDAOSTest.groovy:541]:
```suggestion
                    stage('Build RPM on Leap 15.5') {
```

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: vars/getChrootName.groovy (hunk 1 above)
Comment (phender) [reply vars/getChrootName.groovy]: Fixed

File: vars/packageBuildingPipelineDAOSTest.groovy (hunk 2 above)
Comment (phender) [reply vars/packageBuildingPipelineDAOSTest.groovy]: Fixed

File: vars/packageBuildingPipelineDAOSTest.groovy (hunk 3 above)
Comment (phender) [reply vars/packageBuildingPipelineDAOSTest.groovy]: Fixed

File: vars/packageBuildingPipelineDAOSTest.groovy (hunk 4 above)
Comment (phender) [reply vars/packageBuildingPipelineDAOSTest.groovy]: Fixed

File: vars/packageBuildingPipelineDAOSTest.groovy (hunk 6 above)
Comment (phender) [reply vars/packageBuildingPipelineDAOSTest.groovy]: Fixed

File: vars/packageBuildingPipelineDAOSTest.groovy (hunk 7 above)
Comment (phender) [reply vars/packageBuildingPipelineDAOSTest.groovy]: Fixed

File: vars/packageBuildingPipelineDAOSTest.groovy (hunk 8 above)
Comment (phender) [reply vars/packageBuildingPipelineDAOSTest.groovy]: Fixed

**Comment by phender** [comment]
The only failure in https://build.hpdd.intel.com/job/daos-stack/job/pipeline-lib/job/PR-450/3/ was an NLT failure in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/ci-daos-stack-pipeline-lib-PR-450-release-2.6/5/testReport/ which should not be related to this change and is most likely related to https://daosio.atlassian.net/browse/DAOS-16787

**Review by brianjmurrell — APPROVED** [review]

**Pull request merged**